datafusion/physical-optimizer/src/optimizer.rs

```diff
@@ -145,11 +146,9 @@ impl PhysicalOptimizer {
             // are not present, the load of executors such as join or union will be
             // reduced by narrowing their input tables.
             Arc::new(ProjectionPushdown::new()),
-            Arc::new(EnsureCooperative::new()),
-            // This FilterPushdown handles dynamic filters that may have references to the source ExecutionPlan.
-            // Therefore it should be run at the end of the optimization process since any changes to the plan may break the dynamic filter's references.
-            // See `FilterPushdownPhase` for more details.
-            Arc::new(FilterPushdown::new_post_optimization()),
+            // ReverseOrder: Detect DESC sorts that can use reverse scan
+            // This marks reverse_scan=true on DataSourceExec
+            Arc::new(PushdownSort::new()),
             // The SanityCheckPlan rule checks whether the order and
```

Comment:
this comment I think needs to be moved down too (it is no longer with the sanity check plan implementation)